conan/internal/api/install/generators.py

```diff
@@ -118,18 +118,20 @@ def write_generators(conanfile, app):
             with conanfile_exception_formatter(conanfile, "generate"):
                 conanfile.generate()
 
-    if conanfile.virtualbuildenv:
-        mkdir(new_gen_folder)
-        with chdir(new_gen_folder):
-            from conan.tools.env.virtualbuildenv import VirtualBuildEnv
-            env = VirtualBuildEnv(conanfile)
-            env.generate()
-    if conanfile.virtualrunenv:
-        mkdir(new_gen_folder)
-        with chdir(new_gen_folder):
-            from conan.tools.env import VirtualRunEnv
-            env = VirtualRunEnv(conanfile)
-            env.generate()
+    if envs_generation is None:
+        if conanfile.virtualbuildenv:
+            mkdir(new_gen_folder)
+            with chdir(new_gen_folder):
+                from conan.tools.env.virtualbuildenv import VirtualBuildEnv
+                env = VirtualBuildEnv(conanfile)
+                # TODO: Check length of env.vars().keys() when adding NotEmpty
+                env.generate()
+        if conanfile.virtualrunenv:
+            mkdir(new_gen_folder)
+            with chdir(new_gen_folder):
+                from conan.tools.env import VirtualRunEnv
+                env = VirtualRunEnv(conanfile)
+                env.generate()
 
     _generate_aggregated_env(conanfile)
```

Comment:
There is one scenario that this will still produce env-files: if the build system is for example msvc+ninja, the ``conanvcvars`` to activate the Visual prompt will be there, and ``conanbuild.bat`` will be generated. is this expected? It is true that in certain conditions like using VSCode, the IDE can also manage vcvars, so this might be included in the ``if envs_generation is None``?